### PR TITLE
IBX-8657: Reworked CDP security layer to not rely on deprecated guard authenticators

### PR DESCRIPTION
**Pull request by konradoboza**
| :ticket: Issue | IBX-8657 |
|----------------|-----------|

#### Related PRs: 
- https://github.com/ibexa/cdp/pull/27

#### Description:
This time changes are pretty simple - we only replace deprecated `guard` configuration key with our own, adjusted CDP authenticator implementation.

#### For QA:
The whole flow related to integration with CDP needs to be verified.

#### Documentation:
Config change.

### CODE REVIEW

File: ibexa/commerce/5.0/config/packages/security.yaml [hunk 1]
```diff
@@ -83,8 +83,8 @@ security:
         # Uncomment `ibexa_cdp` rule if you are using Ibexa CDP connector
         #ibexa_cdp:
         #    pattern: /cdp/webhook
```
Comment (Steveb-p) [inline ibexa/commerce/5.0/config/packages/security.yaml:85]: I didn't notice it before, but I assume this pattern should only match if route **ends** with `/cdp/webhook`?
```suggestion
        #    pattern: /cdp/webhook$
```

File: ibexa/commerce/5.0/config/packages/security.yaml [hunk 2]
```diff
@@ -83,8 +83,8 @@ security:
...
-        #    guard:
-        #        authenticator: 'Ibexa\Cdp\Security\CdpRequestAuthenticator'
+        #    custom_authenticators:
+        #        - 'Ibexa\Cdp\Security\CdpRequestAuthenticator'
```
Comment (Steveb-p) [inline ibexa/commerce/5.0/config/packages/security.yaml:87]: Does this endpoint work in context of other firewalls? Can this be accessed with a different authentication that `CdpRequestAuthenticator` (like currently logged in user)?

If no, then this should probably have (and be) it's own [entrypoint](https://symfony.com/doc/current/security/entry_point.html)

File: ibexa/commerce/5.0/config/packages/security.yaml (hunk 1 above)
Comment (konradoboza) [reply ibexa/commerce/5.0/config/packages/security.yaml]: As discussed internally, brought back unused `Ibexa\Cdp\Security\RequestMatcher` and made sure it's being used, ref: https://github.com/ibexa/recipes-dev/pull/136/commits/8c928fab22ef08fcca28617655e25e47d22da339.

File: ibexa/commerce/5.0/config/packages/security.yaml (hunk 2 above)
Comment (konradoboza) [reply ibexa/commerce/5.0/config/packages/security.yaml:87]: According to Symfony doc:
> [!NOTE]
> Starting with Symfony 5.2, the custom authenticator is automatically registered as an entry point if it implements AuthenticationEntryPointInterface.
> Prior to 5.2, you had to configure the entry point separately using the entry_point option. Read [The Entry Point: Helping Users Start Authentication](https://symfony.com/doc/5.x/security/entry_point.html) for more information.

**Review by Steveb-p — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Comment by konradoboza** [comment]
Merging as it's required by already merged `ibexa/cdp` counterpart PR.

**Pull request merged**